### ISSUE TITLE
[TW-1435] Ability to sync the full history of tasks including completed ones

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#1462]
_Renato Alves on 2014-10-16T13:50:02Z says:_ 

 One can use 
```
task sync init
```
to sync all pending tasks.

But there is currently no way to sync complete/deleted tasks.

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2014-10-16T13:50:02Z
Modified: 2016-01-23T19:39:29Z
```

**Comment by taskwarrior** [comment]
_Renato Alves on 2014-10-16T13:50:52Z says:_ 

 This should also help circumvent the "Common ancestor not found" problem sometimes caused by partial upload of task history.

**Comment by taskwarrior** [comment]
_Renato Alves on 2015-01-15T14:56:32Z says:_ 

 Paul Beckingham on IRC:

```
By the way, I'm thinking that "task sync init" should take an optional "task sync init all" arg, which could upload everything, but not be the default.
Also, "task sync init reset" would maybe wipe the data.
```

**Comment by matt-snider** [comment]
Hi, is it still not possible to backup the entire history via taskserver?

**Comment by pbeckingham** [comment]
This is is marked as closed in 2.5.1.  It is implemented.

**Comment by matt-snider** [comment]
Thanks for the response. In some cases deleting tx.data on the server and running sync init was resulting in an empty tx.data file, so it looked a lot like this issue. Somehow it ended up working after a few more tries. Not sure how but maybe related to the data corruption/locking issue. 

On July 29, 2019 10:16:25 PM UTC, Paul Beckingham <notifications@github.com> wrote:
>This is is marked as closed in 2.5.1.  It is implemented.

--
Best regards, 
Matt Snider
